src/gpg/open.py

```diff
@@ -0,0 +1,60 @@
+import gpg
+import os
+import sys
+import time
+import textwrap
+
+
+def print_signatures(verify_result):
+    c = gpg.Context()
+    for signature in verify_result.signatures:
+        user = c.get_key(signature.fpr).uids[0].uid
+        fpr = signature.fpr
+        signed_time = time.ctime(signature.timestamp)
+
+        message = '''
+                  Good signature from "{user}"
+                  with key {fingerprint}
+                  made at {time}
+                  '''.format(user=user,
+                             fingerprint=fpr,
+                             time=signed_time)
+
+        print(textwrap.dedent(message))
+
+
+def open_file(sealed_file_path, output_file_path):
+    """
+    sign and encrypt file_path to the recipents
+    """
+
+    try:
+        c = gpg.Context()
+        with open(sealed_file_path, "rb") as cfile:
+            try:
+                plaintext, result, verify_result = c.decrypt(
+                    cfile, verify=True)
+
+                if os.environ['DEBUG'] == 'yes':
+                    print(verify_result, result)
+
+                print_signatures(verify_result)
+
+            except gpg.errors.BadSignatures:
+                print("Could not verify signatures")
+                exit(3)
+
+        with open(output_file_path, "wb") as nfile:
+            nfile.write(plaintext)
```

Comment:
Can these 2 lines be written like this?
```
nfile = open(output_file_path, "wb")
nfile.write(plaintext)
```

If yes, then the `with` construct seems a bit redundant in this case.

Reply:
Similar but not exact they just differ by a close statement. `with` constructs guarantees that files are closed after the with gets over. It's just a nice way to depict that a particular resource will be closed after this and will only be used in `with`'s boundaries. 
Yes for small no. of operations the one you suggested (and adding a close at the end) is also fine but this helps in larger ones. And I prefer it as it is one line less :-) .